src/mlpack/core/data/save.hpp

```diff
@@ -130,7 +128,7 @@ bool Save(const std::string& filename,
           const std::string& name,
           T& t,
           const bool fatal = false,
-          format f = format::autodetect,
+          FileType f = FileType::AutoDetect,
```

Comment:
I think we need to keep the old `format` type for reverse compatibility: `format::json` is not the same as `FileType::JSON`.  That's a little frustrating for sure, but I don't see another way.  Of course we can use the expanded `FileType` for the `DataOptions` `Load()` versions, and when mlpack 5 comes we can remove this overload and life gets a lot easier.